src/Umbraco.Web/Routing/UrlProvider.cs

```diff
@@ -210,17 +208,20 @@ public string GetUrl(int id, UrlProviderMode mode)
         /// </remarks>
         public string GetUrl(int id, Uri current, UrlProviderMode mode)
         {
-            var url = _urlProviders.Select(provider => provider.GetUrl(_umbracoContext, id, current, mode))
+            string url = _urlProviders.Select(provider => provider.GetUrl(_umbracoContext, id, current, mode))
                 .FirstOrDefault(u => u != null);
             return url ?? "#"; // legacy wants this
         }
 
         internal string GetUrlFromRoute(int id, string route)
         {
-            var provider = _urlProviders.OfType<DefaultUrlProvider>().FirstOrDefault();
-            var url = provider == null 
-                ? route // what else?
-                : provider.GetUrlFromRoute(route, UmbracoContext.Current, id, _umbracoContext.CleanedUmbracoUrl, Mode);
+            IUrlProvider provider = _urlProviders.OfType<IUrlProvider>().FirstOrDefault();

```

Comment:
No need to check/convert the type, as `_urlProviders` is an `IEnumerable<IUrlProvider>`:
```suggestion
            IUrlProvider provider = _urlProviders.FirstOrDefault();
```